mantidproject/mantid
# Correctly populate ties when loading function from string in fitbrowser

**Description of work.**

The ties between parameters are not stored on the individual functions in a composite function as they can tie parameter across functions (unlike fixing a parameter or applying constraints). Previous code attempted to apply ties when initiating the parameter properties of the child functions, here I initiate the ties separately after the child functions browsers have been initiated. To help do this I have made public a composite function method that parses the prefix to return the function index - I thought this would be harmless and I didn't want to reproduce it.

I have removed the old code in `initParameters` that tried to apply the ties, as far as I know that branch is never entered because ties are always on the composite function - I can always replace it (it doesn't get in the way this fix).

I have put in a test in python (which seems to be where the tests for fitting live) - I check the function string loaded matches the functions string of the composite function in the browser (which is true even on master), and that the ties have been set correctly on the child property browsers (to do this I made and exposed to python a function to check the member variable m_ties isn't empty).

I think this is quite an old bug so it is not necessary to have in release 6.1

**To test:**
(1) Follow instruction in the issue
(2) Try with more complicated functions such as
`name=Gaussian,Height=100,PeakCentre=1.45125,Sigma=0.210207;name=Lorentzian,Amplitude=100,PeakCentre=3,FWHM=3;ties=(f1.Amplitude=f0.Height,f1.FWHM=f1.PeakCentre)`
(3) Remove ties, add more ties - check there is no exception
(4) Run a fit - the ties should be respected

Fixes #31552






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
